hs_core/hydroshare/resource.py

```diff
@@ -354,11 +359,13 @@ def create_resource(
 
         fed_zone_home_path = ''
         if fed_res_path:
-            fed_zone_home_path = fed_res_path
             resource.resource_federation_path = fed_res_path
+            fed_zone_home_path = fed_res_path
             resource.save()
-        elif fed_res_file_names:
-            fed_zone_home_path = utils.get_federated_zone_home_path(fed_res_file_names[0])
+
+        # TODO: this option should be eliminated. Require explicit zone path
+        elif len(source_names) > 0:
```

Comment:
This behavior is problematic because the user can pass polymorphic zone paths and there is no check that they are in the same zone. An alternative is to explicitly check that the zones are all the same.